src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/ConfigCommands/ConfigRunners.cs

```diff
@@ -136,12 +136,12 @@ public static string GetValueForConfigKey(ISettings settings, string key, bool s
         /// <summary>
         /// Logs each section of the configuration settings that will be applied, grouped by file path.
         /// </summary>
-        public static void LogSectionsWithPaths(Dictionary<string, VirtualSettingSection> sections, ILogger logger)
+        public static void LogSectionsWithPaths(IEnumerable<string> sections, Settings settings, ILogger logger)
         {
             foreach (var section in sections)
             {
-                logger.LogMinimal(section.Key + ":");
-                var items = section.Value.Items;
+                logger.LogMinimal(section + ":");
+                var items = settings.GetSection(section).Items;
```

Comment:
We must guard against the `settings.GetSection()` being null because that code explicitly returns null as a possibility.